pdns/validate-recursor.cc

```diff
@@ -69,14 +74,19 @@ vState validateRecords(const vector<DNSRecord>& recs)
     state = getKeysFor(sro, recs.begin()->d_name, keys); // um WHAT DOES THIS MEAN - try first qname??
    
     LOG("! state = "<<vStates[state]<<", now have "<<keys.size()<<" keys "<<endl);
+    
+    g_stats.dnssecResults[state]++;
     return state;
   }
   
   LOG("Took "<<sro.d_queries<<" queries"<<endl);
-  if(validrrsets.size() == cspmap.size()) // shortcut - everything was ok
+  if(validrrsets.size() == cspmap.size()) {// shortcut - everything was ok
+    g_stats.dnssecResults[Secure]++;
     return Secure;
+  }
 
   if(keys.empty()) {
+    g_stats.dnssecResults[Insecure]++;
     return Insecure;
   }
```

Comment:
I see like 6 instances of the ++ plus return - wonder if we can refactor that? a macro maybe?